syntax = "proto3";
package arktouros.proto.collector.v1;

option java_multiple_files = true;
option java_package = "edu.npu.arktouros.proto.collector.v1";
option java_outer_classname = "SpanServiceProto";

import "span/span.proto";

service SpanService {
  rpc Export(SpanRequest) returns (SpanResponse);
}

message SpanRequest {
  repeated arktouros.proto.span.v1.Span spans = 1;
}

message SpanResponse {
  // The number of rejected log records.
  //
  // A `rejected_<signal>` field holding a `0` value indicates that the
  // request was fully accepted.
  int64 rejected_span_records = 1;

  // A developer-facing human-readable message in English. It should be used
  // either to explain why the server rejected parts of the data during a partial
  // success or to convey warnings/suggestions during a full success. The message
  // should offer guidance on how users can address such issues.
  //
  // error_message is an optional field. An error_message with an empty value
  // is equivalent to it not being set.
  string error_message = 2;
}